rosidl_generator_py/resource/_msg.py.template

```diff
@@ -93,30 +93,32 @@ class @(spec.base_type.type)(metaclass=Metaclass):
     def __init__(self, **kwargs):
         assert all(['_' + key in self.__slots__ for key in kwargs.keys()]), \
             "Invalid arguments passed to constructor: %r" % kwargs.keys()
-@[for field in spec.fields]@
-@[if field.default_value]@
+@[  for field in spec.fields]@
+@[    if field.default_value]@
         self.@(field.name) = kwargs.get(
             '@(field.name)', @(spec.base_type.type).@(field.name.upper())__DEFAULT)
-@[else]@
-@[if not field.type.is_primitive_type()]@
+@[    else]@
+@[      if not field.type.is_primitive_type()]@
         from @(field.type.pkg_name).msg import @(field.type.type)
-@[end if]@
-@[if field.type.type == 'byte']@
+@[      end if]@
+@[      if field.type.type == 'byte']@
         self.@(field.name) = kwargs.get('@(field.name)', b'0')
-@[elif field.type.type == 'char']@
+@[      elif field.type.type in [
+            'char',
+            'string']]@
```

Comment:
Sorry this is still work in progress, I'm trying to figure out why the strings passed to C don't have the null character at the end (on the C side). This is highly experimental and not ready for review.
I'm curious though, why should a char be initialized to '\0' but not a string ?